pandas/tseries/tools.py

```diff
@@ -410,21 +434,82 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
                 raise e
 
     if arg is None:
-        return arg
-    elif isinstance(arg, tslib.Timestamp):
-        return arg
+        return None
+
+    # handle origin
+    if origin == 'julian':
+
+        original = arg
+        j0 = tslib.Timestamp(0).to_julian_date()
+        if unit != 'D':
+            raise ValueError("unit must be 'D' for origin='julian'")
+        try:
+            arg = arg - j0
+        except:
+            raise ValueError("incompatible 'arg' type for given "
+                             "'origin'='julian'")
+
+        # premptively check this for a nice range
+        j_max = tslib.Timestamp.max.to_julian_date() - j0
+        j_min = tslib.Timestamp.min.to_julian_date() - j0
+        if np.any(arg > j_max) or np.any(arg < j_min):
+            raise tslib.OutOfBoundsDatetime(
+                "{original} is Out of Bounds for "
+                "origin='julian'".format(original=original))
+
+    elif origin not in ['unix', 'julian']:
+
+        # arg must be a numeric
+        original = arg
+        if not ((is_scalar(arg) and (is_integer(arg) or is_float(arg))) or
+                is_numeric_dtype(np.asarray(arg))):
+            raise ValueError(
+                "'{arg}' is not compatible with origin='{origin}'; "
+                "it must be numeric with a unit specified ".format(
+                    arg=arg,
+                    origin=origin))
+
+        # we are going to offset back to unix / epoch time
+        try:
+            offset = tslib.Timestamp(origin) - tslib.Timestamp(0)
+        except tslib.OutOfBoundsDatetime:
+            raise tslib.OutOfBoundsDatetime(
+                "origin {} is Out of Bounds".format(origin))
+        except ValueError:
+            raise ValueError("origin {} cannot be converted "
+                             "to a Timestamp".format(origin))
+
+        # convert the offset to the unit of the arg
+        # this should be lossless in terms of precision
+        offset = offset // tslib.Timedelta(1, unit=unit)
+
+        arg = np.asarray(arg)
```

Comment:
Wouldn't it be simpler to check if it is a list and then convert to array? (or check that is is not a scalar/series/index) 

Then the ``arg = arg + offset`` does just what you want, and the conversion back as below is not needed